main.cpp

```diff
@@ -92,6 +92,123 @@ good to go!
  */
 
 #include <iostream>
+
+ /*4) make them return the correct primitive type. e.g. if you're implementing the FloatType::add function, your implementation would start like this:
+        float FloatType::add( float lhs, float rhs ) */
+
+#include <math.h>
+
+
+///////// FLOAT  ...............
+///////// 
+struct FloatType
+{
+    float add(float lhs, float rhs );
+    float subtract(float lhs, float rhs );
+    float multiply(float lhs, float rhs );
+    float divide(float lhs, float rhs );
+};
+
+float FloatType::add(float lhs, float rhs)
+{
+    return lhs + rhs;
+}
+
+float FloatType::subtract(float lhs, float rhs)
+{
+    return lhs - rhs;
+}
+
+float FloatType::multiply(float lhs, float rhs)
+{
+    return lhs * rhs;
+}
+
+float FloatType::divide(float lhs, float rhs)
+{
+    if ( rhs == 0.f )
+    {
+        std::cout <<std::endl<<"warning, floating point division by zero returns 'inf' !"<<std::endl;
+        return lhs / rhs;
+    }
+    return lhs / rhs;
+}
+
+
+
+///////// DOUBLE ...............
+///////// 
+struct DoubleType 
+{
+    double add(double lhs, double rhs);
+    double subtract(double lhs, double rhs);
+    double multiply(double lhs, double rhs);
+    double divide(double lhs, double rhs);
+};
+
+double DoubleType::add(double lhs, double rhs)
+{
+    return lhs + rhs;
+}
+
+double DoubleType::subtract(double lhs, double rhs)
+{
+    return lhs - rhs;
+}
+
+double DoubleType::multiply(double lhs, double rhs)
+{
+    return lhs * rhs;
+}
+double DoubleType::divide(double lhs, double rhs)
+{
+    if ( rhs == 0.0 )
+    {
+        std::cout <<std::endl<<"warning, floating point division by zero returns 'inf' !" <<std::endl;
+        return lhs / rhs;
+    }
+    return lhs / rhs;
```

Comment:
You need to add space around the `<<` so the code is READABLE.
When you share code with other people, you need to make sure that it is as easy to read as possible.
This includes the formatting of the code.  You want the viewers to be able to look at your code and not get stuck figuring out what you wrote because your code is smashed together and difficult to read.
Treat it like you're writing sentences.   You put spaces between every word for a reason: Readability and understanding. 
wedon'twritesentenceslikethisbecauseitwouldmakeitreallyhardtoread.
So don't write code like that.
```
std::cout << "this is an expression" << " " << "with gaps between the <<" << std::endl;
```

You have 2 identical return statements.
Don't Repeat Yourself. 
D.R.Y. 
Revise your logic so you only have 1 return statement.
- [ ] revise